lib/code_corps_web/controllers/user_controller.ex

```diff
@@ -28,64 +28,81 @@ defmodule CodeCorpsWeb.UserController do
     conn |> render("index.json-api", data: users)
   end
 
-  @spec show(Conn.t, map) :: Conn.t
+  @spec show(Conn.t(), map) :: Conn.t()
   def show(%Conn{} = conn, %{"id" => id}) do
     with %User{} = user <- User |> Repo.get(id) |> preload() do
       conn |> render("show.json-api", data: user)
     end
   end
 
-  @spec create(Conn.t, map) :: Conn.t
+  @spec create(Conn.t(), map) :: Conn.t()
   def create(%Conn{} = conn, %{} = params) do
-    with {:ok, %User{} = user} <- %User{} |> User.registration_changeset(params) |> Repo.insert(),
-         user <- preload(user)
-    do
-      conn |> put_status(:created) |> render("show.json-api", data: user)
+    with {:ok, %User{} = user} <- params |> Accounts.create() do
+      case user |> Map.get(:claimed_invites) do
+        user_invites when is_list(user_invites) ->
+          user_invites |> Enum.map(fn invite ->
+            user.id |> Analytics.SegmentTracker.track("Claimed User Invite", invite)
+          end)
+
+        _other ->
+          nil
+      end
```

Comment:
Possibly we can extract this into a `maybe_track_invite_claim`, but other than that, not sure how to improve on this. Note that user creation itself is tracked implicitly via a plug. We should switch that as well at some point, to add clarity.